apps/services/documents/src/app/modules/document-provider/controllers/provider.controller.ts

```diff
@@ -6,38 +9,50 @@ import {
   Param,
   Post,
   Put,
+  UseGuards,
 } from '@nestjs/common'
 import {
   ApiCreatedResponse,
   ApiHeader,
   ApiOkResponse,
   ApiTags,
 } from '@nestjs/swagger'
+import { environment } from '../../../../environments'
 import { DocumentProviderService } from '../document-provider.service'
 import { CreateProviderDto } from '../dto/createProvider.dto'
 import { UpdateProviderDto } from '../dto/updateProvider.dto'
 import { Provider } from '../models/provider.model'
-import { NationalId } from '../utils/nationalId.decorator'
 
+const namespace = `${environment.audit.defaultNamespace}/providers`
+
+@UseGuards(IdsUserGuard)
 @ApiTags('providers')
 @ApiHeader({
   name: 'authorization',
   description: 'Bearer token authorization',
 })
 @Controller('providers')
+@Audit({ namespace })
 export class ProviderController {
   constructor(
     private readonly documentProviderService: DocumentProviderService,
+    private readonly auditService: AuditService,
   ) {}
 
   @Get()
   @ApiOkResponse({ type: [Provider] })
-  async getAllProviders(): Promise<Provider[] | null> {
-    return await this.documentProviderService.getProviders()
+  @Audit<Provider[]>({
+    resources: (providers) => providers.map((provider) => provider.id),
+  })
+  async getAllProviders(): Promise<Provider[]> {
+    return this.documentProviderService.getProviders()
   }
 
   @Get(':id')
   @ApiOkResponse({ type: Provider })
+  @Audit<Provider>({
+    resources: (provider) => provider?.id,
+  })
```

Comment:
Can `provider` be null/undefined? (also below)

Reply:
No, as these two actions check for a falsy object, and throw an error if it is not found. 
So I removed the optional operator.